staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -242,3 +267,72 @@ func maybeTruncateUserAgent(req *http.Request) string {
 
 	return ua
 }
+
+// copyWithoutManagedFields will make a deep copy of the specified object and
+// will discard the managed fields from the copy.
+// The specified object is expected to be a meta.Object or a "list".
+// The specified object obj is treated as readonly and hence not mutated.
+// On return, an error is set if the function runs into any error while
+// removing the managed fields, the boolean value is true if the copy has
+// been made successfully, otherwise false.
+func copyWithoutManagedFields(obj runtime.Object) (runtime.Object, bool, error) {
+	isAccessor := true
+	if _, err := meta.Accessor(obj); err != nil {
+		isAccessor = false
+	}
+	isList := meta.IsListType(obj)
+	_, isTable := obj.(*metav1.Table)
+	if !isAccessor && !isList && !isTable {
+		return nil, false, nil
+	}
+
+	// TODO a deep copy isn't really needed here, figure out how we can reliably
+	//  use shallow copy here to omit the manageFields.
+	copy := obj.DeepCopyObject()
+
+	if isAccessor {
+		if err := removeManagedFields(copy); err != nil {
+			return nil, false, err
+		}
+	}
+
+	if isList {
+		if err := meta.EachListItem(copy, removeManagedFields); err != nil {
+			return nil, false, err
+		}
+	}
+
+	if isTable {
+		table := copy.(*metav1.Table)
+		for i := range table.Rows {
+			rowObj := table.Rows[i].Object
```

Comment:
can `rowObj` ever be empty? (by general convention it should not be).  maybe add a check `if rowObj == nil { continue }`?

Reply:
looks like this comment is outstanding

Reply:
`rawObj` can't be `nil` - it's the `Object` field of type `runtime.RawExtension` from `TableRow`
```
type TableRow struct {
	Cells []interface{} `json:"cells"`
	Conditions []TableRowCondition `json:"conditions,omitempty"`
	Object runtime.RawExtension `json:"object,omitempty"`
}
```

I have added an empty  `TableRow` object to my test to ensure it does not panic.